pkg/secretless/entrypoint/start_secretless.go

```diff
@@ -0,0 +1,125 @@
+package entrypoint
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/cyberark/secretless-broker/internal"
+	secretlessLog "github.com/cyberark/secretless-broker/internal/log"
+	"github.com/cyberark/secretless-broker/internal/plugin/v1/event_notifier"
+	"github.com/cyberark/secretless-broker/pkg/secretless"
+	"github.com/cyberark/secretless-broker/pkg/secretless/config"
+	v2 "github.com/cyberark/secretless-broker/pkg/secretless/config/v2"
+
+	"github.com/cyberark/secretless-broker/internal/plugin"
+	"github.com/cyberark/secretless-broker/internal/profile"
+	"github.com/cyberark/secretless-broker/internal/signal"
+)
+
+// CLIParams holds the command line flag information that Secretless was started
+// with.
+type CLIParams struct {
+	ConfigFile          string
+	ConfigManagerSpec   string
+	DebugEnabled        bool
+	FsWatchEnabled      bool
+	PluginChecksumsFile string
+	PluginDir           string
+	ProfilingMode       string
+	ShowVersion         bool
+}
+
+// StartSecretless method is the main entry point into the broker after the CLI
+// flags have been parsed
+func StartSecretless(params *CLIParams) {
+
+	ShowVersion(params.ShowVersion)
+
+	VerifyPlugins(params.PluginDir, params.PluginChecksumsFile)
+
+	// Construct the deps of Secretless
+	cfg := ReadConfig(params.ConfigFile)
+	logger := secretlessLog.New(params.DebugEnabled)
+	evtNotifier := event_notifier.New(nil)
+	availPlugins := &internal.AvailPluginStub{}
+
+	// Prepare Secretless
+	secretless := internal.NewSecretless(cfg, availPlugins, logger, evtNotifier)
+	signal.StopOnExitSignal(secretless)
+
+	StartPerformanceProfiling(params.ProfilingMode)
```

Comment:
The naming here is a bit misleading. This function conditionally starts performance profiling, I think it might be clearer to have an if statement.

Reply:
Ha.  I went back and forth on "the cleanness of all the logic in a func" vs "the naming being perfectly accurate".  I also considered something like `HandlePerformanceProfiling` which makes no false promise of starting -- what do you think of that?  Open to other suggestions as well.... I would like to keep the logic in the func if possible.

Reply:
I went ahead and changed to `HandlePerformanceProfiling` for now, but lmk if you don't like this solution.